scripts/helpers.py

```diff
@@ -565,3 +566,62 @@ def locate_bus(
         return gdf_co[gdf_co.geometry == min(gdf_co.geometry, key=(point.distance))][
             col
         ].item()  # looks for closest one shape=node
+
+
+def get_last_commit_message():
+    """
+    Function to get the last Git commit message
+    Returns
+    -------
+    result : string
+    """
+    try:
+        # Run the Git command to get the last commit message
+        result = subprocess.run(
+            ["git", "log", "-1", "--pretty=format:%H %s"],
+            capture_output=True,
+            text=True,
+        )
+        return result.stdout.strip()
+    except Exception as e:
+        logging.warning(f"Error getting the last commit message: {e}")
+        return ""
+
+
+def get_submodule_commit_message():
+    """
+    Function to get the last PyPSA-Earth Git commit message
+    Returns
+    -------
+    result : string
+    """
+    try:
+        # Retrieve the last commit message for pypsa-earth
+        submodule_path = f"pypsa-earth"
+        last_commit_message = (
+            subprocess.check_output(
+                ["git", "log", "-n", "1", "--pretty=format:%H %s"],
+                cwd=submodule_path,
+                stderr=subprocess.STDOUT,
+            )
+            .decode()
+            .strip()
+        )
+        return last_commit_message
+    except subprocess.CalledProcessError as e:
+        logging.warning(f"Error executing Git: {e}")
+        return None
+
+
+# Function to update the YAML file with the last commit message as a comment
+def update_config(config):
+    """
+    Function to add the last commit to the config
+    Returns
+    -------
+    config : dict
+    """
+    # Insert the last commit message to config
+    config.update({"git_commit": get_last_commit_message()})
+    config.update({"submodule_commit": get_submodule_commit_message()})
+    return config
```

Comment:
As in the pypsa-earth, I'd recommend to revise the name to make the name more explicit.
There is no need for the comment before the function as there is the docstring.

Moreover, similarly to pypsa-earth, it may be interesting to directly load the function into the snakemake at the beginning, right after the config initialization and processing.
This would be quite interesting as the commit will automatically be available in every script and automatically saved anywhere we do `n.meta = config`

What do you think?

Very good :D